### PR TITLE
Fix for Musixmatch multi-part lyrics

### PR DESCRIPTION
**Pull request by Ubiquitine**
## Description

Sometimes Musixmatch lyrics come in 2 parts.
Also, sometime non-existent lyrics return some content

### Lyrics: Improve Musixmatch backend
* Fix for lyrics that divided into 2 or more <div> elements.
* Ignore "fake" lyrics when lyrics not found but some content still being retrieved.

### CODE REVIEW

File: beetsplug/lyrics.py [hunk 1]
```diff
@@ -291,14 +291,25 @@ def fetch(self, artist, title):
...
+        html_parts = html.split('<p class="mxm-lyrics__content')
+        # Sometimes lyrics come in 2 parts
+        if len(html_parts) > 1:
+            html_part1 = html_parts[-2]
+            lyrics_part1 = extract_text_between(html_part1, '>', '</p>')
+        else:
+            lyrics_part1 = ''
+        html_part2 = html_parts[-1]
+        lyrics_part2 = extract_text_between(html_part2, '>', '</p>')
+        lyrics = lyrics_part1 + '\n' + lyrics_part2
```
Comment (sampsyo) [inline beetsplug/lyrics.py:303]: Would it be simpler to do this with a loop over all the parts? Something roughly like this:

```py
for html_part in html_parts:
    lyrics_parts.append(extract_text_between(html_part1, '>', '</p>'))
lyrics = '\n'.join(lyrics_parts)
```

**Review by sampsyo — COMMENTED** [review]
Looks perfectly reasonable; thanks!! Here's a suggestion for how to simplify the code. Also, could you please add a quick changelog entry?

File: beetsplug/lyrics.py (hunk 1 above)
Comment (Ubiquitine) [reply beetsplug/lyrics.py]: Thanks for the correction.
Indeed, this approach is more elegant. I've updated my branch to use the loop, as well as better non-existing lyrics detection.
I've tested it on couple songs I have, and it seems to work.

**Comment by sampsyo** [comment]
Awesome; looks great! If you don't mind adding a quick changelog entry, I think this is good to go.

**Comment by Ubiquitine** [comment]
I added changelog alongside the description. Or is there other place for this?

**Comment by wisp3rwind** [comment]
> I added changelog alongside the description. Or is there other place for this?

I'm not seeing any changelog entry in this PR; did you actually push it? In any case, the changelog is at https://github.com/beetbox/beets/blob/master/docs/changelog.rst

**Comment by Ubiquitine** [comment]
> > I added changelog alongside the description. Or is there other place for this?
> 
> I'm not seeing any changelog entry in this PR; did you actually push it? In any case, the changelog is at https://github.com/beetbox/beets/blob/master/docs/changelog.rst
Updated changelog.rst

**Comment by sampsyo** [comment]
Looks great to me; thanks!!

**Pull request merged**